boards/xenvm/Kconfig.defconfig

```diff
@@ -6,4 +6,7 @@ if BOARD_XENVM
 config BUILD_OUTPUT_BIN
 	default y
 
+config HEAP_MEM_POOL_SIZE
```

Comment:
> Due to HWMv2 defconfig inheritance the setting must be moved to `Kconfig.defconfig`.

Technically this is a bit imprecise, as
- `xenvm_defconfig` could explicitly set  `CONFIG_HEAP_MEM_POOL_SIZE=16384`
- `xenvm_xenvm_gicv3_defconfig` could override by explicitly set `CONFIG_HEAP_MEM_POOL_SIZE=0`

That is possible because `KERNEL_MEM_POOL` is default `y`.

but as we now get rid of a `_defconfig` file, then this change is a bit nice.

Reply:
> * `xenvm_xenvm_gicv3_defconfig` could override by explicitly set `CONFIG_HEAP_MEM_POOL_SIZE=0`

True, I just didn't want to set a sticky value for it, because the original didn't.